api/comments.rb

```diff
@@ -27,16 +27,34 @@
   sub_comment.anonymous_to_peers = bool_anonymous_to_peers || false
   sub_comment.author = user
   sub_comment.comment_thread = comment.comment_thread
+  sub_comment.children_count = 0
   sub_comment.save
   if sub_comment.errors.any?
     error 400, sub_comment.errors.full_messages.to_json
   else
-    user.subscribe(comment.comment_thread) if bool_auto_subscribe
-    sub_comment.to_hash.to_json
+    comment.set :children_count, comment.children_count + 1
+    if comment.errors.any?
+      error 400, comment.errors.full_messages.to_json
+    else
+      user.subscribe(comment.comment_thread) if bool_auto_subscribe
+      sub_comment.to_hash.to_json
+    end
   end
 end
 
 delete "#{APIPREFIX}/comments/:comment_id" do |comment_id|
+  parent_id = comment.parent_id
   comment.destroy
+  unless parent_id.nil?
+    begin
+      parent_comment = Comment.find(parent_id)
+      if parent_comment.children_count.nil?
+        set_comment_children_count(parent_comment)
+      end
+      parent_comment.set :children_count, parent_comment.children_count - 1
```

Comment:
Couple of issues:
1. If the count wasn't set, you compute the count after the destroy, but the you subtract 1.  Wouldn't this need to be an else condition?
2. As mentioned above, there is a race condition.  So, if we need to store this count, we should compute each time and forget about the subtraction.